src/arch/xtensa/CMakeLists.txt

```diff
@@ -0,0 +1,376 @@
+	endif()
+
+	add_custom_target(
+		bootloader_dump
+		COMMAND cp bootloader bootloader_${fw_name}
+		COMMAND ${CMAKE_OBJCOPY} -O binary ${PROJECT_BINARY_DIR}/src/platform/${platform_folder}/boot_module mod-boot-${fw_name}.bin
+		COMMAND ${CMAKE_OBJCOPY} --add-section .module=mod-boot-${fw_name}.bin --set-section-flags .module=load,readonly bootloader_${fw_name}
+		COMMAND ${CMAKE_OBJCOPY} -O binary bootloader bootloader_${fw_name}.bin
+		COMMAND ${CMAKE_OBJDUMP} -h -D bootloader > bootloader_${fw_name}.map
+		COMMAND ${CMAKE_OBJDUMP} -S bootloader > bootloader_${fw_name}.lst
+		COMMAND ${CMAKE_OBJDUMP} -D bootloader > bootloader_${fw_name}.dis
+		DEPENDS bootloader boot_module
+		VERBATIM
+		USES_TERMINAL
+	)
+
+	set(bootloader_binary_path bootloader_${fw_name})
+
+	add_custom_target(
+		process_base_module
+		COMMAND ${CMAKE_OBJCOPY} -O binary ${PROJECT_BINARY_DIR}/src/platform/${platform_folder}/base_module mod-${fw_name}.bin
+		COMMAND ${CMAKE_OBJCOPY} --add-section .module=mod-${fw_name}.bin --set-section-flags .module=load,readonly sof_${fw_name}
+		DEPENDS prepare_fw_copy_for_rimage base_module bootloader_dump
+		VERBATIM
+		USES_TERMINAL
+	)
+else()
+	set(bootloader_binary_path "")
+
+	add_custom_target(process_base_module DEPENDS prepare_fw_copy_for_rimage)
+endif()
+
+add_custom_target(
+	sof_dump
+	COMMAND ${CMAKE_OBJDUMP} -S sof_${fw_name} > sof_${fw_name}.lst
+	COMMAND ${CMAKE_OBJDUMP} -D sof_${fw_name} > sof_${fw_name}.dis
+	DEPENDS process_base_module
+	VERBATIM
+	USES_TERMINAL
+)
+
+if(MEU_PATH)
+	execute_process(
+		COMMAND ${MEU_PATH}/meu -ver
+		COMMAND grep "Version:"
+		COMMAND cut -d " " -f6
+		OUTPUT_VARIABLE MEU_VERSION
+		OUTPUT_STRIP_TRAILING_WHITESPACE
+	)
+
+	if(MEU_VERSION VERSION_LESS 12.0.0.1035)
+		set(MEU_OFFSET 1152)
+	else()
+		set(MEU_OFFSET 1088)
+	endif()
+
+	set(otc_private_key ${PROJECT_SOURCE_DIR}/rimage/keys/otc_private_key.pem)
+
+	add_custom_target(
+		run_rimage
+		COMMAND ${PROJECT_BINARY_DIR}/rimage_ep/build/rimage
+			-o sof_${fw_name}.ri
+			-p sof_${fw_name}.ldc
+			-m ${fw_name}
+			${bootloader_binary_path}
+			sof_${fw_name}
+			-s ${MEU_OFFSET}
+			-k ${otc_private_key}
+		DEPENDS sof_dump
+		VERBATIM
+		USES_TERMINAL
+	)
+
+	add_custom_target(
+		run_meu
+		COMMAND ${MEU_PATH}/meu -w ./ -s sof_${fw_name}
+			-key ${MEU_PRIVATE_KEY}
+			-stp /usr/bin/openssl
+			-f ${MEU_PATH}/generic_meu_conf.xml
+			-mnver 0.0.0.0 -o sof_${fw_name}.ri
+		DEPENDS run_rimage
+		VERBATIM
+		USES_TERMINAL
+	)
+else()
+	add_custom_target(
+		run_rimage
+		COMMAND ${PROJECT_BINARY_DIR}/rimage_ep/build/rimage
+			-o sof_${fw_name}.ri
+			-p sof_${fw_name}.ldc
+			-m ${fw_name}
+			${bootloader_binary_path}
+			sof_${fw_name}
+		DEPENDS sof_dump
+		VERBATIM
+		USES_TERMINAL
+	)
+	add_custom_target(run_meu DEPENDS run_rimage)
+endif()
+
+add_custom_target(
+	bin
+	COMMAND cp sof_${fw_name}.ri ${PROJECT_BINARY_DIR}/sof_${fw_name}.ri
+	DEPENDS run_meu
+	VERBATIM
+	USES_TERMINAL
+)
+
+if(CONFIG_BUILD_VM_ROM)
```

Comment:
Please check the platform and compiler used here.
To avoid wrong use and build all use chaos.

Reply:
Just change config

Reply:
Then it should be set in Kconfig files (not cmake) that BUILD_VM_ROM depends on HAVE_ROM and for example platform APOLLOLAKE selects HAVE_ROM. However for me it looks that it's not something bound to platform, more like it's not supported by our CI/qemu/something external yet. Or am I wrong?